File: ruby/update-A-host.rb
#!/usr/bin/ruby

# The FreeDns.io project
# This sample updates an A record to IP address 8.8.8.8

require 'net/http'
require 'uri'

data = {
    'username' => 'foo',
    'password' => '***',
    'host' => 'myhost',
    'record' => 'A',
    'value'=> '8.8.8.8'
}

uri = URI('https://freedns.io/request')
res = Net::HTTP.post_form(uri, data)

#res.body will contain the response 
#res.code will contain the return the HTML status code

File: python/update-A-host.py
#/usr/bin/env python3

"""
     The FreeDns.io project
     This sample updates an A record to IP address 8.8.8.8
"""

from urllib.parse import urlencode
from urllib import request

url = 'https://freedns.io/request'
params = {
  'username': 'foo',
  'password': '***',
  'host': 'myhost',
  'record': 'A',
  'value': '8.8.8.8'
}

data = urlencode(params)
req = request.Request(url, data.encode('utf-8'))

try:
  response = request.urlopen(req)
  content = response.read()
  print(content)
except request.URLError as e:
  print("Error: %d %s (%s)" % (e.code, e.reason, e.read()))
